chain-storage/src/lib.rs

```diff
@@ -1069,11 +798,145 @@ pub mod tests {
+            .unwrap();
+
+        let mut block = genesis_block.make_child(None);
+
+        main_branch_blocks.push(genesis_block);
+
+        for _i in 1..MAIN_BRANCH_LEN {
+            let block_info = BlockInfo::new(
+                block.id.serialize_as_vec(),
+                block.parent.serialize_as_vec(),
+                block.chain_length,
+            );
+            store
+                .put_block(&block.serialize_as_vec(), block_info)
+                .unwrap();
+            main_branch_blocks.push(block.clone());
+            block = block.make_child(None);
+        }
+
+        let mut second_branch_blocks = vec![main_branch_blocks[BIFURCATION_POINT].clone()];
+
+        block = main_branch_blocks[BIFURCATION_POINT].make_child(None);
+
+        for _i in 1..SECOND_BRANCH_LEN {
+            let block_info = BlockInfo::new(
+                block.id.serialize_as_vec(),
+                block.parent.serialize_as_vec(),
+                block.chain_length,
+            );
+            store
+                .put_block(&block.serialize_as_vec(), block_info)
+                .unwrap();
+            second_branch_blocks.push(block.clone());
+            block = block.make_child(None);
+        }
+
+        let expected_tips = {
+            let mut hs = HashSet::new();
+            hs.insert(main_branch_blocks.last().unwrap().id.serialize_as_vec());
+            hs.insert(second_branch_blocks.last().unwrap().id.serialize_as_vec());
+            hs
+        };
+        let actual_tips = HashSet::from_iter(store.get_tips_ids().unwrap().into_iter());
+        assert_eq!(expected_tips, actual_tips);
+
+        store
+            .prune_branch(&second_branch_blocks.last().unwrap().id.serialize_as_vec())
+            .unwrap();
+
+        assert_eq!(
+            vec![main_branch_blocks.last().unwrap().id.serialize_as_vec()],
+            store.get_tips_ids().unwrap()
+        );
+
+        store
+            .get_block(&second_branch_blocks[0].id.serialize_as_vec())
+            .unwrap();
+
+        for i in 1..SECOND_BRANCH_LEN {
+            let block_result = store.get_block(&second_branch_blocks[i].id.serialize_as_vec());
+            assert!(matches!(block_result, Err(Error::BlockNotFound)));
+        }
+    }
+
+    #[test]
+    fn get_blocks_by_chain_length() {
+        const N_BLOCKS: usize = 5;
+
+        let file = tempfile::TempDir::new().unwrap();
+        let mut store = BlockStore::new(file.path()).unwrap();
+
+        let genesis_block = Block::genesis(None);
+        let genesis_block_info = BlockInfo::new(
+            genesis_block.id.serialize_as_vec(),
+            genesis_block.parent.serialize_as_vec(),
+            genesis_block.chain_length,
+        );
+        store
+            .put_block(&genesis_block.serialize_as_vec(), genesis_block_info)
+            .unwrap();
+
+        let mut blocks = vec![];
+
+        for _i in 0..N_BLOCKS {
+            let block = genesis_block.make_child(None);
```

Comment:
Add a test with forks in too

Reply:
This code create 5 different branches actually. Could you please clarify what do you mean?